src/en/space-station-14/core-tech/proposals/appearancesystem-compliance.md

```diff
@@ -0,0 +1,108 @@
+
+### Proposed Solution, Overview
+
+In a perfect world, the process of setting and copying an entity's sprite would be the following:
+
+Setting a sprite: 
+1) An entity is spawned in and has a default prototype with sprite data set; this is effectively the "base" to start from.
+2) A game logic system (`ExampleSystem` & `ExampleComponent`) performs its functions and, as a result, requires the sprite to change.
+3) The game logic system sets a key-value pair in the entity's `AppearanceComponent.AppearanceData`, and ensures the entity has `ExampleVisualizerComponent`. This raises the `OnAppearanceDataChange` event.
+4) The `ExampleVisualizerSystem`, which subscribes to `OnAppearanceDataChanged` for any entity with an `ExampleVisualizerComponent` on account of inheriting `VisualizerSystem`, reads the event.
+5) `ExampleVisualizerSystem` applies sprite changes via its logic based on the `AppearanceData` as input.
+
+Copying a sprite:
+1) An entity (_A_) wants to copy the exact visuals of another entity (_B_).
+2) _A_ looks up the prototype for _B_ and copies the SpriteComponent data from the prototype.
+3) _A_ queries all `VisualizerSystem` components from entity _B_ and applies them to itself.
+4) _A_ copies all `AppearanceData` from _B_, which triggers any relevant `VisualizerSystem` to perform their logic, setting sprites in `SpriteSystem`.
+
+This would result in a flow of operations that looks like this:
+![image](https://github.com/user-attachments/assets/3ffcd550-d56a-4cbf-a9b6-ec3b43b7fc21)
+
+Any change to `SpriteSystem` must go through `VisualizerSystem`s first, which in turn only act on data from `AppearaceSystem`. 
+__You should be able to derive the exact appearance of an entity based on VisualizerSystems and AppearanceData alone.__
+
+In comparison, this is how the flow of operations looks currently:
+![image](https://github.com/user-attachments/assets/ebd0aaa8-ba73-4fdd-9585-45b5fc9ff594)
+
+The red arrows show "destructive" operations where you either can't know for certain whether an edit to a sprite has been made, or it requires iterating over all systems for all objects to be fully certain that you have copied visuals fully.
+
+### Proposed Solution, implementation
+
+To achieve the overview, the following implementation would be required:
+
+- Any game logic system that modifies sprites is split up into a game logic system and a visualizer system (i.e. inherits `VisualizerSystem`). The game logic system *sets* `AppearanceData` for the visualizer system to work with, instead of modifying the sprites directly.
+- All visualizer systems are connected to visualizer components (e.g. `ClothingSystem` have `ClothingVisualizerSystem` and `ClothingVisualizerComponent`). The visualizer system handles the visual logic, and the component on an entity indicates an entity should run the system.
+- `SpriteSystem`/`SpriteComponent` may not be accessed outside of visualizer systems to prevent game logic systems modifying sprites.
```

Comment:
If anything, this should just be a convention / rule of thumb, and not actually enforced via an analyser or whatever because there's probably always going to be some exceptions. Especially when it comes to systems/components that control animated sprite layers, where you don't want to dirty/network unnecessary data every tick (e.g., something like `RgbLightControllerSystem`)